presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -83,7 +106,24 @@ public void onSuccess(BaseResponse<TaskInfo> result)
                             @Override
                             public void onFailure(Throwable t)
                             {
-                                log.error("TaskInfoCallback failed %s", t);
+                                // record failure
+                                try {
+                                    errorTracker.requestFailed(t);
+                                }
+                                catch (PrestoException e) {
+                                    stop();
+                                    lastException.set(e);
+                                    return;
+                                }
+                                ListenableFuture<?> errorRateLimit = errorTracker.acquireRequestPermit();
+                                try {
+                                    // synchronously wait on throttling
+                                    errorRateLimit.get(maxErrorDuration.toMillis(), TimeUnit.MILLISECONDS);
+                                }
+                                catch (InterruptedException | ExecutionException | TimeoutException e) {
+                                    // throttling error is not fatal, just log the error.
```

Comment:
Why is it not fatal? What kind of error do we expect here?

Reply:
Discussed offline, we reach to here because we got throttled - updateScheduledExecutor.scheduleWithFixedDelay will schedule task periodically in sequence, if the previous task is >blocking (synchronously wait), it won't schedule the next one.